torch/testing/_internal/mps_utils.py

```diff
@@ -0,0 +1,680 @@
+    ),
+    "prod": MPSSkipInfo(
+        NONCONTIGUOUS,
+        dtypes=[torch.complex64],
+    ),
+    "put": MPSSkipInfo(NONCONTIGUOUS),
+    "qr": MPSSkipInfo(NONCONTIGUOUS),
+    "quantile": MPSSkipInfo(
+        NONCONTIGUOUS,
+        dtypes=[torch.float32],
+        upper=15.0,
+        lower=14.0,
+    ),
+    "rand_like": MPSSkipInfo(NONCONTIGUOUS),
+    "randint_like": MPSSkipInfo(NONCONTIGUOUS),
+    "randn_like": MPSSkipInfo(NONCONTIGUOUS),
+    "renorm": MPSSkipInfo(
+        NONCONTIGUOUS,
+        dtypes=[torch.complex64],
+    ),
+    "repeat": MPSSkipInfo(
+        NONCONTIGUOUS,
+        dtypes=[torch.complex64],
+    ),
+    "round": MPSSkipInfo(
+        NONCONTIGUOUS,
+        variant="decimals_0",
+    ),
+    "rsub": MPSSkipInfo(NONCONTIGUOUS),
+    "scatter_add": MPSSkipInfo(
+        NONCONTIGUOUS,
+        dtypes=[torch.complex64],
+    ),
+    "scatter": MPSSkipInfo(
+        NONCONTIGUOUS,
+        dtypes=[torch.complex64],
+    ),
+    "scatter_reduce": [
+        MPSSkipInfo(
+            NONCONTIGUOUS,
+            variant="amax",
+            dtypes=[torch.int64],
+        ),
+        MPSSkipInfo(
+            NONCONTIGUOUS,
+            variant="amin",
+            dtypes=[torch.int64],
+        ),
+    ],
+    "sigmoid": MPSSkipInfo(
+        NONCONTIGUOUS,
+        dtypes=[torch.int64],
+    ),
+    "sinc": MPSSkipInfo(NONCONTIGUOUS),
+    "softmax": MPSSkipInfo(
+        NONCONTIGUOUS,
+        variant="with_dtype",
+    ),
+    "special.airy_ai": MPSSkipInfo(NONCONTIGUOUS),
+    "special.bessel_j0": MPSSkipInfo(NONCONTIGUOUS),
+    "special.bessel_j1": MPSSkipInfo(NONCONTIGUOUS),
+    "special.bessel_y0": MPSSkipInfo(NONCONTIGUOUS),
+    "special.bessel_y1": MPSSkipInfo(NONCONTIGUOUS),
+    "special.chebyshev_polynomial_t": MPSSkipInfo(NONCONTIGUOUS),
+    "special.chebyshev_polynomial_u": MPSSkipInfo(NONCONTIGUOUS),
+    "special.entr": MPSSkipInfo(NONCONTIGUOUS),
+    "special.erfcx": MPSSkipInfo(NONCONTIGUOUS),
+    "special.hermite_polynomial_h": MPSSkipInfo(NONCONTIGUOUS),
+    "special.hermite_polynomial_he": MPSSkipInfo(NONCONTIGUOUS),
+    "special.i0e": MPSSkipInfo(NONCONTIGUOUS),
+    "special.i1e": MPSSkipInfo(NONCONTIGUOUS),
+    "special.laguerre_polynomial_l": MPSSkipInfo(NONCONTIGUOUS),
+    "special.log_ndtr": MPSSkipInfo(NONCONTIGUOUS),
+    "special.modified_bessel_i0": MPSSkipInfo(NONCONTIGUOUS),
+    "special.modified_bessel_i1": MPSSkipInfo(NONCONTIGUOUS),
+    "special.modified_bessel_k0": MPSSkipInfo(NONCONTIGUOUS),
+    "special.modified_bessel_k1": MPSSkipInfo(NONCONTIGUOUS),
+    "special.ndtri": MPSSkipInfo(NONCONTIGUOUS),
+    "special.polygamma": MPSSkipInfo(NONCONTIGUOUS),
+    "special.scaled_modified_bessel_k0": MPSSkipInfo(NONCONTIGUOUS),
+    "special.scaled_modified_bessel_k1": MPSSkipInfo(NONCONTIGUOUS),
+    "special.spherical_bessel_j0": MPSSkipInfo(NONCONTIGUOUS),
+    "special.xlog1py": MPSSkipInfo(NONCONTIGUOUS),
+    "special.zeta": MPSSkipInfo(NONCONTIGUOUS),
```

Comment:
It feels like those should be skipped with `unimplemented` reason, shouldn't they?